src/server/services/buzz.service.ts

```diff
@@ -345,3 +345,41 @@ export async function completeStripeBuzzTransaction({
     throw error;
   }
 }
+
+export async function refundTransaction(
+  transactionId: string,
+  description?: string,
+  details?: MixedObject
+) {
+  const body = JSON.stringify({
+    description,
+    details,
+  });
+
+  const response = await fetch(`${env.BUZZ_ENDPOINT}/transactions/${transactionId}/refund`, {
+    method: 'POST',
+    headers: { 'Content-Type': 'application/json' },
+    body,
+  });
+
+  if (!response.ok) {
+    const cause: { reason: string } = JSON.parse(await response.text());
+
+    switch (response.status) {
+      case 400:
+        throw throwBadRequestError(cause.reason, cause);
+      case 409:
+        throw throwBadRequestError('There is a conflict with the transaction', cause);
+      default:
+        throw new TRPCError({
+          code: 'INTERNAL_SERVER_ERROR',
+          message: 'An unexpected error ocurred, please try again later',
+          cause,
+        });
+    }
+  }
```

Comment:
Seems like this can be ported over to a separate method and be reused, no need to do it now, you can leave a note for later

Reply:
yeah we should do that, incidentally i noticed a weird maybe-bug with this. the error that came back looked off.